table/tables/tables_test.go

```diff
@@ -392,3 +392,105 @@ func (ts *testSuite) TestTableFromMeta(c *C) {
+	tk.MustExec("USE test_hidden;")
+	tk.MustExec("CREATE TABLE hidden (a int primary key, b int as (a+1), c int, d int as (c+1), e int, f tinyint as (a+1));")
+	tk.MustExec("insert into hidden values (1, default, 3, default, 5, default);")
+	tb, err := ts.dom.InfoSchema().TableByName(model.NewCIStr("test_hidden"), model.NewCIStr("hidden"))
+	c.Assert(err, IsNil)
+	colInfo := tb.Meta().Columns
+	// Set column b, d, f to hidden
+	colInfo[1].Hidden = true
+	colInfo[3].Hidden = true
+	colInfo[5].Hidden = true
+	tc := tb.(*tables.TableCommon)
+	// Reset related caches
+	tc.VisibleColumns = nil
+	tc.WritableColumns = nil
+	tc.HiddenColumns = nil
+
+	// Basic test
+	cols := tb.VisibleCols()
+	c.Assert(table.FindCol(cols, "a"), NotNil)
+	c.Assert(table.FindCol(cols, "b"), IsNil)
+	c.Assert(table.FindCol(cols, "c"), NotNil)
+	c.Assert(table.FindCol(cols, "d"), IsNil)
+	c.Assert(table.FindCol(cols, "e"), NotNil)
+	hiddenCols := tb.HiddenCols()
+	c.Assert(table.FindCol(hiddenCols, "a"), IsNil)
+	c.Assert(table.FindCol(hiddenCols, "b"), NotNil)
+	c.Assert(table.FindCol(hiddenCols, "c"), IsNil)
+	c.Assert(table.FindCol(hiddenCols, "d"), NotNil)
+	c.Assert(table.FindCol(hiddenCols, "e"), IsNil)
+
+	// Can't select with b and d and f
+	tk.MustQuery("select * from hidden;").Check(testkit.Rows("1 3 5"))
+	_, err = tk.Exec("select b from hidden;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'field list'")
+	_, err = tk.Exec("select d from hidden;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'd' in 'field list'")
+	err = tk.QueryToErr("select a, c, e from hidden;")
+	c.Assert(err, IsNil)
+	_, err = tk.Exec("select d from hidden;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'd' in 'field list'")
+	_, err = tk.Exec("select * from hidden where b>1;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'where clause'")
+	_, err = tk.Exec("select * from hidden order by b;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'order clause'")
+	_, err = tk.Exec("select * from hidden group by b;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'group statement'")
+
+	// Can't update with b and d
+	_, err = tk.Exec("update hidden set d=1;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'd' in 'field_list'")
+	_, err = tk.Exec("update hidden set a=1 where b=1;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'where clause'")
+	_, err = tk.Exec("update hidden set a=1 where c=2 order by b;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'order clause'")
+	tk.MustExec("update hidden set a = 5;")
+	colInfo[5].Hidden = false
+	tc.VisibleColumns = nil
+	tc.WritableColumns = nil
```

Comment:
Forget `tc.HiddenColumns = nil` ?

Reply:
We don't need to reset `tc.HiddenColumns` here since we don't use `HiddenCols` in next tests.
Besides, it seem that we don't need reset `tc.WritableColumns` too.